docs/source-2.0/glossary.rst

```diff
@@ -0,0 +1,152 @@
+        if a member is optional or always present.
+
+    Projection
+        A specific view of a Smithy model that has added, removed, or
+        transformed model components.
+
+        .. seealso:: :doc:`guides/building-models/build-config`
+
+    Reserved words
+        Identifiers and words that cannot be used in a
+        :term:`target environment`. Reserved words can be contextual or global
+        to the target language (for example, a word might only reserved when
+        used as a structure property but not when used as the name of a
+        shape). Code generators are expected to automatically translate
+        reserved words into an identifier that is safe to use in the
+        target environment.
+
+    Runtime libraries
+        The libraries used at runtime in a :term:`target environment`.
+        For example, HTTP clients, type implementations like
+        big decimal, etc.
+
+    Semantic model
+        The Smithy semantic model is an in-memory representation the shapes,
+        traits, and metadata defined in the Smithy model. In Smithy's
+        reference implementation, the semantic model is contained in the
+        `Model class`_.
+
+    Serde
+        Shortened version of serialization and deserialization.
+
+    Service closure
+        The shapes connected to a service. These shapes are code generated.
+
+    Shapes
+        Shapes are named declarations of Smithy types that make up the
+        :term:`semantic model`.
+
+    Smithy-Build
+        A model transformation framework built on top of the
+        :term:`Smithy reference implementation`. Code generators are
+        implemented as :ref:`smithy-build <smithy-build>` plugins.
+
+    smithy-build.json
+        The file used to configure Smithy-Build. Code generators are configured
+        and executed by adding plugins to smithy-build.json files in various
+        projections.
+
+        .. seealso:: :doc:`guides/building-models/build-config`
+
+    Smithy model
+        Smithy models define services, operations, resources, and shapes.
+        Smithy models are made up of one or more files to form the
+        semantic model. Model files can use a JSON or IDL representation.
+
+    Smithy reference implementation
+        The Java implementation of Smithy that is used to load, validate,
+        transform, and extract information from Smithy models.
+
+    Smithy type
+        The types of shapes that can be defined in a Smithy model (for example,
+        string, integer, structure, etc.).
+
+    Symbol
+    Symbols
+        The qualified name of a type in a target programming language. Symbols
+        are used to map Smithy shapes to types in a :term:`target environment`,
+        refer to language types, and refer to libraries that might be needed by
+        the generated code. A symbol contains an optional namespace, optional
+        namespace delimiter, name, a map of additional properties, a
```

Comment:
nit: move the additional properties callout to the end of the list since it's a grabbag.